shared/subscribers/service.go

```diff
@@ -82,16 +97,30 @@ func (s *Service) ObserveEvents(ctx context.Context, retries int, eTypes []event
 			case bs := <-in:
 				// batch size changed: drain buffer and send data
 				data = append(data, sub.UpdateBatchSize(ctx, bs)...)
-				if len(data) > 0 {
+				dataLength := len(data)
+
+				if dataLength > 0 {
 					trySend()
 				}
+
+				if dataLength > s.maxBufferSize {
```

Comment:
What if `trySend` actually managed to send the data? It makes more sense to me to check `dataLength` _before_ trying to send

Reply:
if it can send the data then by definition its not a slow consumer, so checking before doesn't make sense

Reply:
fair point, switched them around, though if we get that close to the buffer limit pretty much nailed on that the consume is slow